lib/Listener/GroupUserRemovedListener.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace OCA\Photos\Listener;
+
+use OCA\Photos\Album\AlbumMapper;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\Group\Events\UserRemovedEvent;
+
+class GroupUserRemovedListener implements IEventListener {
+	private AlbumMapper $albumMapper;
+
+	public function __construct(AlbumMapper $albumMapper) {
+		$this->albumMapper = $albumMapper;
+	}
+
+	public function handle(Event $event): void {
+		if (!($event instanceof UserRemovedEvent)) {
+			return;
+		}
+        
+        // Get all shared albums for this group:
+        $albums_group = $this->albumMapper->getSharedAlbumsForCollaboratorWithFiles($event->getGroup()->getGID(), AlbumMapper::TYPE_GROUP);
+        // Get all albums shared with this specific user:
+        $albums_user = $this->albumMapper->getSharedAlbumsForCollaboratorWithFiles($event->getUser()->getUID(), AlbumMapper::TYPE_USER);
```

Comment:
Same:

- You have the list of albums for a group
- For each album:
    - If the use is not in the collaborator list, remove its files.